src/MountedTraversal.js

```diff
@@ -177,6 +178,12 @@ export function parentsOfInst(inst, root) {
   return pathToNode(inst, root).reverse();
 }
 
+export function instMatchesObjectProps(inst, props) {
+  if (!isDOMComponent(inst)) return false;
+  const node = getNode(inst);
+  return !isEmpty(props) && isMatch(propsOfNode(node), props);
```

Comment:
First and foremost it needs to be correct, and performance is the least important (not that it should be ignored). The API should do something sensible in all cases.

Thinking about it, however, I think that if, in fact, `propsOfNode` returns an empty object, then `instMatchesObjectProps` *should* return true. So, perhaps removing the isEmpty call is the right thing.